### DISCUSSION TITLE
Find binary name of formulae

### DISCUSSION THREAD

**Discussion by quackduck** [Homebrew/discussions#450] [Tap maintenance and Homebrew development] [answered]
In short: Is there a way to get the name of the binary a formula installs?

Hello! I'm writing [a program](https://github.com/quackduck/tii) that does what `command-not-found` (on Ubuntu) does but for macOS. 

On most Linux distros, when a command is not found, a helpful message showing what package manager commands can be run to install it are shown. This is not available in macOS, presumably because it lacks a default package manager. Most developers however use (the fantastic) Homebrew.

For this reason, I made [a program](https://github.com/quackduck/tii) that is triggered when a command is not found and looks for Homebrew formulae matching the name of the command. It does this by simply making a list of the names of all the Ruby scripts (found in `/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/`).

However, there is one shortcoming. The program relies on formulae names, and not the names of the binaries they install. I was not able to find any way to get the binary name (except the inefficient way of searching for `bin.install` within the files). Is there a way to do this?

**Comment by SMillerDev** [comment, marked as answer]
There isn't and even `bin.install` won't help you since that is only used if upstream doesn't provide a makefile or something similar to move the files to `bin`

**Reply by quackduck** [reply]
So in the makefile case, symlinks are not used, correct?

**Reply by SMillerDev** [reply]
There will be symlinks from `#{prefix}/bin` to `#{cellar}/bin`

**Reply by quackduck** [reply]
Wait so the makefile has to install the symlinks? Who's making the symlinks here? When you're using `bin.install`, Homebrew's making the symlinks. In makefiles, who?

**Reply by SMillerDev** [reply]
`bin.install` doesn't create symlinks. It just moves files to the `bin` directory of the Cellar. Anything that is there will then be symlinked by brew to the general `bin` as part of the install process.

**Reply by quackduck** [reply]
Ah okay. What about the makefile scenario?

**Reply by SMillerDev** [reply]
In that case make installs them into the Cellar and brew links the Cellar as it always would.

**Comment by jonchang** [comment]
It sounds like you're looking for https://github.com/Homebrew/homebrew-command-not-found
